src/shared/remoted_op.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 23, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "remoted_op.h"
+
+#ifndef WIN32
+
+/**
+ * @brief Looks for the OS architecture in a string. Possibles architectures
+ *        are x86_64, i386, i686, sparc, amd64, ia64, AIX, armv6, armv7.
+ *        The function will return a pointer to allocated memory that must
+ *        be de-allocated by the caller.
+ *
+ * @param[in] os_header String that contains the architecture. Usually uname.
+ * @retval A string pointer to the architecture.
+ */
+char * get_os_arch(char * os_header) {
+    const char * ARCHS[] = { "x86_64", "i386", "i686", "sparc", "amd64", "ia64", "AIX", "armv6", "armv7", NULL };
+    char * os_arch = NULL;
+    int i;
+
+    for (i = 0; ARCHS[i]; i++) {
+        if (strstr(os_header, ARCHS[i])) {
+            os_strdup(ARCHS[i], os_arch);
+            break;
+        }
+    }
+
+    if (!ARCHS[i]) {
+        os_strdup("", os_arch);
+    }
+
+    mdebug2("Detected architecture from %s: %s", os_header, os_arch);
+    return os_arch;
+}
+
+/**
+ * @brief Parses an OS uname string. All the OUT parameters are pointers
+ *        to allocated memory that must be de-allocated by the caller.
+ *
+ * @param[in] msg The agent update message string to be parsed.
+ * @param[Out] os_name The OS name.
+ * @param[Out] os_major The OS major version.
+ * @param[Out] os_minor The OS minor version.
+ * @param[Out] os_build The OS build number.
+ * @param[Out] os_version The OS full version.
+ * @param[Out] os_codename The OS codename.
+ * @param[Out] os_platform The OS platform.
+ * @param[Out] os_arch The OS architecture.
+ */
```

Comment:
We should include an `@pre` sentence about every output pointer must be valid.

Reply:
All the output pointers were removed.